[BUG] OIDC authentication failed

OIDC authentication failed when using PoketID as the OIDC provider.

MeshMonitor v2.4.4


**Docker Compose Config**

```
services:
  main_app:
    container_name: meshmonitor
    environment:
      - ALLOWED_ORIGINS=https://meshmonitor.domain.local,http://ip_address:3067
      - COOKIE_SECURE=true
      - MESHTASTIC_NODE_IP=ip_address
      - NODE_ENV=production
      - OIDC_CLIENT_ID=clientid
      - OIDC_CLIENT_SECRET=secret
      - OIDC_ENABLED=true
      - OIDC_ISSUER=https://pid.domain.local
      - OIDC_REDIRECT_URI=https://meshmonitor.domain.local/api/auth/oidc/callback
      - OIDC_SCOPES=openid profile email
      - SESSION_SECRET=secret
      - TRUST_PROXY=true
      - TZ=America/New_York
    image: ghcr.io/yeraze/meshmonitor:latest
    ports:
      - target: 3001
        published: "3067"
        protocol: tcp
    restart: unless-stopped
    volumes:
      - type: bind
        source: /mnt/Storage1/meshmonitor
        target: /data
    devices: []
    cap_add: []
    network_mode: bridge
    privileged: false
```


**Error in logs**
```
[ERROR] OIDC callback error: ClientError: invalid response encountered
    at e (file:///app/node_modules/openid-client/build/index.js:116:12)
    at errorHandler (file:///app/node_modules/openid-client/build/index.js:139:23)
    at Module.authorizationCodeGrant (file:///app/node_modules/openid-client/build/index.js:954:21)
    at handleOIDCCallback (file:///app/dist/server/auth/oidcAuth.js:86:44)
    ... 5 lines matching cause stack trace ...
    at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
  code: 'OAUTH_INVALID_RESPONSE',
  [cause]: OperationProcessingError: response parameter "iss" (issuer) missing
      at OPE (file:///app/node_modules/oauth4webapi/build/index.js:116:12)
      at Module.validateAuthResponse (file:///app/node_modules/oauth4webapi/build/index.js:2054:15)
      at Module.authorizationCodeGrant (file:///app/node_modules/openid-client/build/index.js:951:42)
      at handleOIDCCallback (file:///app/dist/server/auth/oidcAuth.js:86:44)
      at file:///app/dist/server/routes/authRoutes.js:236:28
      at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17)
      at next (/app/node_modules/router/lib/route.js:157:13)
      at Route.dispatch (/app/node_modules/router/lib/route.js:117:3)
      at handle (/app/node_modules/router/index.js:435:11)
      at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
    code: 'OAUTH_INVALID_RESPONSE',
    [cause]: { parameters: [Object] }
  }
}
[ERROR] OIDC callback error: ClientError: invalid response encountered
    at e (file:///app/node_modules/openid-client/build/index.js:116:12)
    at errorHandler (file:///app/node_modules/openid-client/build/index.js:139:23)
    at Module.authorizationCodeGrant (file:///app/node_modules/openid-client/build/index.js:954:21)
    at handleOIDCCallback (file:///app/dist/server/auth/oidcAuth.js:86:44)
    ... 5 lines matching cause stack trace ...
    at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
  code: 'OAUTH_INVALID_RESPONSE',
  [cause]: OperationProcessingError: response parameter "iss" (issuer) missing
      at OPE (file:///app/node_modules/oauth4webapi/build/index.js:116:12)
      at Module.validateAuthResponse (file:///app/node_modules/oauth4webapi/build/index.js:2054:15)
      at Module.authorizationCodeGrant (file:///app/node_modules/openid-client/build/index.js:951:42)
      at handleOIDCCallback (file:///app/dist/server/auth/oidcAuth.js:86:44)
      at file:///app/dist/server/routes/authRoutes.js:236:28
      at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17)
      at next (/app/node_modules/router/lib/route.js:157:13)
      at Route.dispatch (/app/node_modules/router/lib/route.js:117:3)
      at handle (/app/node_modules/router/index.js:435:11)
      at Layer.handleRequest (/app/node_modules/router/lib/layer.js:152:17) {
    code: 'OAUTH_INVALID_RESPONSE',
    [cause]: { parameters: [Object] }
  }
}
```

--- a/src/server/auth/oidcAuth.ts
+++ b/src/server/auth/oidcAuth.ts
@@ -104,27 +104,29 @@ export async function generateAuthorizationUrl(
  * Handle OIDC callback and create/update user
  */
 export async function handleOIDCCallback(
-  code: string,
-  state: string,
+  callbackUrl: URL,
   expectedState: string,
   codeVerifier: string,
-  expectedNonce: string,
-  redirectUri: string
+  expectedNonce: string
 ): Promise<User> {
   if (!oidcConfig) {
     throw new Error('OIDC not initialized');
   }
 
   try {
+    // Extract state from callback URL for validation
+    const state = callbackUrl.searchParams.get('state');
+
     // Validate state
     if (state !== expectedState) {
       throw new Error('Invalid state parameter');
     }
 
     // Exchange code for tokens
+    // Pass the full callback URL with all parameters (including iss if present)
     const tokenResponse = await client.authorizationCodeGrant(
       oidcConfig,
-      new URL(redirectUri + `?code=${code}&state=${state}`),
+      callbackUrl,
       {
         pkceCodeVerifier: codeVerifier,
         expectedState,

--- a/src/server/routes/authRoutes.ts
+++ b/src/server/routes/authRoutes.ts
@@ -304,16 +304,20 @@ router.get('/oidc/callback', async (req: Request, res: Response) => {
       return res.status(400).send('Invalid session state');
     }
 
-    const redirectUri = getEnvironmentConfig().oidcRedirectUri || `${req.protocol}://${req.get('host')}/api/auth/oidc/callback`;
+    // Construct the full callback URL with all query parameters
+    // This preserves additional parameters like 'iss' (issuer) that some OIDC providers send
+    const protocol = req.protocol;
+    const host = req.get('host');
+    const path = req.path;
+    const queryString = req.url.split('?')[1] || '';
+    const fullCallbackUrl = new URL(`${protocol}://${host}${path}?${queryString}`);
 
     // Handle callback and create/update user
     const user = await handleOIDCCallback(
-      code as string,
-      state as string,
+      fullCallbackUrl,
       expectedState,
       codeVerifier,
-      expectedNonce,
-      redirectUri
+      expectedNonce
     );
 
     // Clear OIDC session data

